backend/api/office_hours/oh_event.py

```diff
@@ -0,0 +1,118 @@
+"""OH Event API
+
+This API is used to access OH Event data."""
+
+from datetime import datetime, timedelta
+from fastapi import APIRouter, Depends
+from backend.models.coworking.time_range import TimeRange
+from backend.models.office_hours.oh_event import OfficeHoursEventDraft
+from backend.models.office_hours.oh_event_details import OfficeHoursEventDetails
+
+from ..authentication import registered_user
+from ...models import User
+
+
+__authors__ = ["Sadie Amato", "Bailey DeSouza"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+api = APIRouter(prefix="/api/office-hours/event")
+openapi_tags = {
+    "name": "Office Hours",
+    "description": "Office hours event functionality",
+}
+
+
+@api.post("", response_model=OfficeHoursEventDetails, tags=["Office Hours"])
+def new_oh_event(
+    oh_event: OfficeHoursEventDraft,
+    subject: User = Depends(registered_user),
+    oh_event_service: OfficeHoursEventService = Depends(),
+) -> OfficeHoursEventDetails:
+    """
+    Adds a new OH event to the database
+
+    Returns:
+        OfficeHoursEventDetails: OH Event created
+    """
+    return oh_event_service.create(subject, oh_event)
+
+
+@api.put("", response_model=OfficeHoursEventDetails, tags=["Academics"])
+def update_oh_event(
+    oh_event: OfficeHoursEventDraft,
+    subject: User = Depends(registered_user),
+    oh_event_service: OfficeHoursEventService = Depends(),
+) -> OfficeHoursEventDetails:
+    """
+    Updates an OfficeHoursEvent to the database
+
+    Returns:
+        OfficeHoursEventDetails: OH Event updated
+    """
+    return oh_event_service.update(subject, oh_event)
+
+
+@api.delete("/{oh_event_id}", response_model=None, tags=["Office Hours"])
+def delete_oh_event(
+    oh_event_id: int,
+    subject: User = Depends(registered_user),
+    oh_event_service: OfficeHoursEventService = Depends(),
+):
+    """
+    Deletes an OfficeHoursEvent from the database
+    """
+    return oh_event_service.delete(subject, oh_event_id)
+
+
+@api.get("/{section_id}", response_model=list[OfficeHoursEventDetails], tags=["Office Hours"])
```

Comment:
Should this be moved to the section API? Even though is lists events, it feels logically like a child route of a section.

Reply:
Great point, I've moved this into the section API!